internal/invocation/stripe_executor.go

```diff
@@ -0,0 +1,95 @@
+package invocation
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+var (
+	// Default values differ from java impl. Also queue size is calculated differently.
+	// Java Client: queueSize per worker = defaultEventQueueCapacity / defaultEventWorkerCount
+	// Go Client: queueSize per worker = defaultEventQueueCapacity
+	defaultEventQueueCapacity = int32(10000)
+	defaultEventWorkerCount   = int32(runtime.NumCPU())
+)
+
+// executor represents the function that will run on workers of stripeExecutor.
+type executor func(queue chan func(), quit chan struct{}, wg *sync.WaitGroup)
+
+// stripeExecutor executes given "tasks" preserving the order among the ones that are given with the same key.
+type stripeExecutor struct {
+	quit       chan struct{}
+	wg         *sync.WaitGroup
+	execFn     executor
+	taskQueues []chan func()
+	queueCount int32
+}
+
+// newStripeExecutor returns a new stripeExecutor with default configuration.
+func newStripeExecutor() stripeExecutor {
+	// ignore error, default values do not raise error.
+	ex, _ := newStripeExecutorWithConf(defaultEventWorkerCount, defaultEventQueueCapacity)
+	return ex
+}
+
+// newStripeExecutor returns a new stripeExecutor with configured queueCount and queueSize.
+func newStripeExecutorWithConf(queueCount, queueSize int32) (stripeExecutor, error) {
```

Comment:
Let's rename this to `newStripeExecutorWithConfig` to be consistent with the rest of the codebase.

Reply:
Sure!

Reply:
Thanks 👍